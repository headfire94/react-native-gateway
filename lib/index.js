export PortalContext from './portal/portal-context'
export PortalEnter from './portal/portal-enter'
export PortalExit from './portal/portal-exit'
export PortalProvider from './portal/portal-provider'
