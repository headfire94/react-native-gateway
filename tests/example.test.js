/**
 * @jest-environment jsdom
 */
import { View } from "react-native";
import { mount } from "enzyme";
import React from "react";

import { PortalEnter, PortalExit, PortalProvider } from "../lib/index";

test("mounts correctly", () => {
  const ToPort = () => <View test-name="to-port" />;
  const App = (
    <PortalProvider>
      <View test-name="top">
        <PortalEnter name="example">
          <ToPort />
        </PortalEnter>
      </View>
      <PortalExit />
    </PortalProvider>
  );
  const wrapper = mount(App);

  expect(wrapper).toMatchSnapshot();
});

test("PortalExit contains ToPort component", () => {
  const ToPort = () => <View test-name="to-port" />;
  const App = (
    <PortalProvider>
      <View test-name="top">
        <PortalEnter name="example">
          <ToPort />
        </PortalEnter>
      </View>
      <PortalExit />
    </PortalProvider>
  );
  const wrapper = mount(App);

  expect(wrapper.find(PortalExit).find(ToPort)).toMatchSnapshot();
});
