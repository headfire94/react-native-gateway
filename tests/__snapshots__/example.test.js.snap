// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mounts correctly 1`] = `ShallowWrapper {}`;

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "red",
    }
  }
/>
`;
